### PR TITLE
Add upload to digital ocean spaces

### PR DESCRIPTION
**Pull request by arjunchib**
Adds support to upload memes to DigitalOcean Spaces. The method for uploading can be set via `UPLOAD` environment variable. Changes the url returned by api to be absolute so the client can use it without any modifications.

### CODE REVIEW

File: src/resolvers/createMeme.js [hunk 1]
```diff
@@ -8,22 +9,24 @@ module.exports = async function({ name, author, url, start, end }, { ip, db }) {
...
-    start: start ? start : 0,
-    end: end ? end : 0
+    url,
+    start,
+    end
   }
 
-  return local_upload(yt_download(meme))
-    .then(function(file_name) {
-      const new_url = '/memes/' + file_name
```
Comment (MatthewHannah) [inline src/resolvers/createMeme.js:22]: What the hell did I put in here to start with...

File: src/meme-upload/local.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const path = require('path')
+const shortid = require('shortid')
+
+function add(stream) {
+  const fileName = shortid.generate() + '.mp3'
+
+  const filePath = path.join(process.env.MEME_DIR, fileName)
+  stream.pipe(fs.createWriteStream(filePath))
+
+  const url = `/memes/${fileName}`
```
Comment (MatthewHannah) [inline src/meme-upload/local.js]: We should unify the url field between local and spaces (i.e. the url in Mongo for local storage should be the fully-qualified name), but I'm not sure how to do that easily.

**Review by MatthewHannah — CHANGES_REQUESTED** [review]
You should also update the yarn seed script to use the proper upload & url combination. Other than that I'm glad the general scheme is pretty extensible.

File: src/meme-upload/local.js (hunk 2 above)
Comment (arjunchib) [reply src/meme-upload/local.js]: Yeah I was thinking about that and totally forgot to change that.

**Pull request merged**